metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,413 @@
+    DatasetProfileClass,
+    DatasetPropertiesClass,
+    DateTypeClass,
+    TimeTypeClass,
+    NullTypeClass,
+    NumberTypeClass,
+    OtherSchemaClass,
+    RecordTypeClass,
+    SchemaFieldDataTypeClass,
+    StringTypeClass,
+    SubTypesClass,
+)
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+# -------------------------------------------------- constants -------------------------------------------------- #
+PLATFORM_NAME_IN_DATAHUB = "cassandra"
+
+# we always skip over ingesting metadata about these keyspaces
+# TODO: make this configurable?
+SYSTEM_KEYSPACE_LIST = set(["system", "system_auth", "system_schema", "system_distributed", "system_traces"])
+
+# this keyspace contains details about the cassandra cluster's keyspaces, tables, and columns
+SYSTEM_SCHEMA_KESPACE_NAME = "system_schema"
+# these are the names of the tables we're interested in querying metadata from
+CASSANDRA_SYSTEM_SCHEMA_TABLES = {
+    "keyspaces": "keyspaces",
+    "tables": "tables",
+    "columns": "columns",
+}
+# these column names are present on the system_schema tables
+CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES = {
+    "keyspace_name": "keyspace_name",
+    "table_name": "table_name",
+    "column_name": "column_name",
+    "column_type": "type",
+}
+
+# -------------------------------------------------- queries -------------------------------------------------- #
+# get all keyspaces
+GET_KEYSPACES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['keyspaces']}"
+# get all tables for a keyspace
+GET_TABLES_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['tables']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s"
+# get all columns for a table
+GET_COLUMNS_QUERY = f"SELECT * FROM {SYSTEM_SCHEMA_KESPACE_NAME}.{CASSANDRA_SYSTEM_SCHEMA_TABLES['columns']} WHERE {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['keyspace_name']} = %s AND {CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES['table_name']} = %s"
+
+# -------------------------------------------------- setup the config and source reporter -------------------------------------------------- #
+
+# config
+class CassandraSourceConfig(PlatformInstanceConfigMixin, EnvConfigMixin):
+    contact_point: str = Field(
+        default="localhost", description="The cassandra instance contact point domain (without the port)."
+    )
+    port: str = Field(
+        default="10350", description="The cassandra instance port."
+    )
+    username: str = Field(
+        default=None, description="The username credential."
+    )
+    password: str = Field(
+        default=None, description="The password credential."
+    )
+    excludeKeyspaces: List[str] = Field(
+        default=[], description="The keyspaces to exclude."
+    )
+
+
+# source reporter
+@dataclass
+class CassandraSourceReport(SourceReport):
```

Comment:
anything else worth adding here? other sources seem to vary in how detailed these get